src/Workspaces/Core/Portable/CodeActions/CodeAction.cs

```diff
@@ -746,5 +746,37 @@ public static NoChangeAction Create(
              => SpecializedTasks.Null<Solution>();
     }
 
+    internal sealed class DualChangeAction : DocumentChangeAction
```

Comment:
should not be needed.  remove.  you should be able to just make a normal code aciton, which no-ops in the ComputePreviewOperations method.